src/main/webapp/app/overview/course-overview.component.ts

```diff
@@ -276,6 +301,32 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
         }
     }
 
+    /** initialize courses attribute by retreiving recently accessed courses from the server */
+    getRecentlyAccessedCourses() {
+        this.courseService.findAllForDashboard().subscribe({
+            next: (res: HttpResponse<CoursesForDashboardDTO>) => {
+                if (res.body) {
+                    const courses: Course[] = [];
+                    for (const courseDto of res.body.courses) {
+                        courses.push(courseDto.course);
+                    }
+                    this.courses = courses.sort((a, b) => (a.title ?? '').localeCompare(b.title ?? ''));
+                    if (this.courses.length > 3) {
+                        const lastAccessedCourseIds = this.courseAccessStorageService.getLastAccessedCourses();
+                        this.courses = this.courses.filter((course) => lastAccessedCourseIds.includes(course.id!));
+                    }
+                }
+            },
+        });
+    }
+
+    /** Navigate to a new Course */
+    switchCourse(course: Course) {
+        this.router.navigate(['courses', course.id]).then(() => {
+            window.location.reload();
+        });
```

Comment:
This is not good due to bad performance and doesn't look good either (flickering screen), as we are reloading the complete page. We should avoid doing this whenever possible.

Not sure if the below is the best solution - you might want to investigate other options
```suggestion
        this.router.navigateByUrl('/', { skipLocationChange: true }).then(() => {
            this.router.navigate(['courses', course.id]);
        });
```

Reply:
agreed